### DISCUSSION TITLE
PMP::clip() does not modify the clipping mesh

### DISCUSSION THREAD

**Discussion by stla** [CGAL/cgal#6641] [Q&A] [answered]
Hello,

I clipped a cube to the volume bounded by a sphere:

![CGALclippedMesh](https://user-images.githubusercontent.com/4466543/171996135-5074cf66-9a16-47a4-ae45-8bf31003485b.png)

It is said in the doc that the clipping mesh (the sphere) is modified when doing `PMP::clip()`. However it is not modified (this is the sphere at right).

**Comment by pentacular** [comment]
It's probably something wrong in the code you've written.

**Reply by stla** [reply]
Do you really think I didn't envisage this possibility before posting? ^^

**Comment by sloriot** [comment]
It is not modified if you pass the option `do_not_modify(true)`

**Reply by stla** [reply]
It is false (sure: if it were true, I couldn't clip to a volume).

**Reply by sloriot** [reply]
Do you have a minimal example, or at least the call to CGAL's function?

**Reply by stla** [reply]
I do:

```cpp
  ...
  const bool clipping = PMP::clip(
    mesh, clipper, PMP::parameters::clip_volume(true),
    PMP::parameters::clip_volume(true).do_not_modify(false)
  );
  mesh.collect_garbage();
  clipper.collect_garbage();
  ...
```

Looks correct? I can prepare a MWE if you want.

**Comment by sloriot** [comment, marked as answer]
Just to be sure we are on the same thing here: the sphere being modified means that the sphere is corefined so the geometry of the clipper does not change. Only its faces are split and re-triangulated.

**Reply by stla** [reply]
Ah indeed, I was expecting a change of the geometry. I will check more carefully.

**Reply by stla** [reply]
You're right, as usual. The mesh is modified.
